### PR TITLE
atc/api: Super admin user first steps

### PR DESCRIPTION
**Pull request by taylorsilva**
tl;dr: Modify endpoint behavior to return everything when the user is an `Admin` i.e. owner of the main team

- We modified the following methods to behave properly for an admin user:
  - ListAllPipelines
  - ListAllJobs
  - ListBuilds
  - ListAllResources
  - ListWorkers
- Added the appropriate tests.
- this is currently functional, that the admin user is able to see all the pipelines when logging in from the web ui.
- fly hits team endpoints, so still not working as per this PR, we will address that in #4195.
- Also, applying actions to pipelines in the web ui (unpause, expose, trigger job), this should be addressed in #4194. 

cc: @youssb 

fixes: #4193

### CODE REVIEW

File: atc/db/resource_factory.go [hunk 1]
```diff
@@ -40,12 +43,28 @@ func (r *resourceFactory) VisibleResources(teamNames []string) ([]Resource, erro
...
+		RunWith(r.conn).
+		Query()
+	if err != nil {
+		return nil, err
+	}
+
+	return r.parseResources(rows)
+}
+
+func (r *resourceFactory) parseResources(resourceRows *sql.Rows) ([]Resource, error) {
```
Comment (pivotal-jwinters) [inline atc/db/resource_factory.go]: Typically we would call this `scanResources` and it would be a package level function (instead of being tied to the resourceFactory). That way it can be reused in other places if needed. See `scanResource` as an example.

File: atc/api/accessor/accessor.go [hunk 2]
```diff
@@ -42,19 +42,23 @@ func (a *access) Claims() jwt.MapClaims {
...
+			if a.hasPermission(teamRole) {
+				return true
 			}
 		}
 	}
 	return false
 }
 
-func (a *access) HasPermission(role string) bool {
+func (a *access) hasPermission(role string) bool {
```
Comment (zoetian) [inline atc/api/accessor/accessor.go]: Just curious, do we have to change this to lower case? Is there some reason to do so?

Comment (cirocosta) [reply atc/api/accessor/accessor.go]: IIRC the reasoning behind the change was that there were no uses for `HasPermissions` outside the scope of the package

**Comment by YoussB** [comment]
@pivotal-jwinters, updated and rebased. Please tell me if there is anything that is missing.

**Comment by YoussB** [comment]
@pivotal-jwinters Will you please reapprove it?
